philipmeneghini/InstaConnect
# Permissions Setup for API

**Summary**
I have created a policy based permission system in the ASP.NET API. Each `UserModel` will have a new `Role` attribute which is used to determine what permissions to apply for each policy used on endpoints. The `Role` class is an `enum` with the following roles:

- Administrator: Has access to all the endpoints
- RegularUser: Most common role. This limits `POST`, `PUT` and `DELETE` on other user's content, comments and personal information
- Guest: Used strictly when a user is creating a new account. This is needed to create a user record as well as send a registration email.
- Unknown: Unknown is a catch all for the rest of the roles. It does not have access to any of the endpoints.
The `JWT` token will hold the role information which is parsed when a user makes a request.

**Testing**

- Verify that administrators can hit all the endpoints while unknown can hit none of them
- Test for specific access when hitting endpoints with the role Regular User and Guest.
- Make sure that none of the normal endpoint functionality has changed.
